functions/src/business-logics/like-logics.ts

```diff
@@ -1,7 +1,100 @@
-// TODO write onLikeLogic
-//  increment post likesCount
-//  add user's userView to post's likes collection
-//  create a notification for post's author
+import {admin, db} from "emberflow/lib";
+import {LogicConfig, LogicResultDoc} from "emberflow/lib/types";
+import {createUserView} from "./utils";
+import {Post, PostView} from "../types";
+
+export const onLikeLogic: LogicConfig = {
+  name: "onLikeLogic",
+  actionTypes: ["create"],
+  modifiedFields: "all",
+  entities: ["like"],
+  logicFn: async (action) => {
+    const {
+      eventContext: {
+        docPath,
+        docId,
+      },
+      user,
+    } = action;
+
+    const postRef = db.doc(docPath);
+    const postDocRef = postRef.parent.parent;
+    if (!postDocRef) {
+      return {
+        name: "onLikeLogic",
+        status: "error",
+        documents: [],
+        message: `Invalid post docpath at ${docPath}`,
+      };
+    }
+
+    const postDocSnapshot = await postDocRef.get();
+    const postDoc = postDocSnapshot.data() as Post;
+    if (!postDoc) {
+      return {
+        name: "onLikeLogic",
+        status: "error",
+        documents: [],
+        message: `Post ${postDocRef.id} does not exist`,
+      };
+    }
+
+    const {
+      "@id": postId,
+      createdBy: {
+        "@id": postAuthorId,
+      },
+    } = postDoc;
+
+    const postLogicResultDoc: LogicResultDoc = {
+      action: "merge",
+      dstPath: postDocRef.path,
+      instructions: {
+        likesCount: "++",
+      },
+    };
+
+    const userView = createUserView(user);
+    const now = admin.firestore.Timestamp.now();
+
+    const likeDoc = {
+      user: userView,
+    };
+
+    const likeDocPath = `posts/${postId}/likes/${docId}`;
+    const likeLogicResultDoc: LogicResultDoc = {
+      action: "create",
+      dstPath: likeDocPath,
+      doc: likeDoc,
```

Comment:
Hi noel, this is the right implementation. 

doc: {...userView}

If we will look at the dbStructure we can see that like entity is a view of user entity, so we are expecting it to have fields equal to the user document. In your case it won't be equal because you put the userView in the user field. This can be solved by just spreading the userView directly in the document